src/main/content/_layouts/doc.html

```diff
@@ -29,31 +29,69 @@
 <!DOCTYPE html>
 <html lang="{{ page.lang | default: site.lang | default: "en" }}">
 
-  {% include head.html %}  
-  <body>
-    
-    {% include analytics.html %}
+    {% include head.html %}
+
+<body>
 
+    {% include analytics.html %}
     {% include header.html %}
-    {% assign t = site.data[site.active_lang].general-reference %}
-    {% assign catsMeta = site.pages | where: 'title', 'doc categories meta' %}
-    {% assign categories = catsMeta[0].categories %}
 
-    {% assign introDoc = site.pages | where: 'title', t.welcome-to-docs %}
+    {% assign t = site.data[site.active_lang].doc %}
+    {% assign url = page.url | split: '/' %}
+    {% assign selected-doc-version = url[2] %}
+
+    {% assign docs-versions = '' | split: ','  | sort %}
+    {% assign docs = site.pages | where: 'layout', 'doc' %}
+    {% assign catsMetaFiles = site.pages | where: 'title', 'doc categories meta' %}
+    {% assign introDocFiles = site.pages | where: 'title', t.welcome-to-docs %} 
+
+    {% for doc in docs %}
+        {% assign dir = doc.dir | split: '/' %}
+        {% if dir[2] %}
+            {% assign docs-versions = docs-versions | push: dir[2] | uniq | sort | reverse %}
+        {% endif%}
+    {% endfor %}
+
+    {% for catsMeta in catsMetaFiles %}
+        {% if catsMeta.dir contains selected-doc-version %}
+            {% assign categories = catsMeta.categories %}
+        {% endif%}
+    {% endfor %}
+
     <div class="container-fluid" id="doc-container">
         <div class="row">
             <div class="col-xl-3" id="toc-col">
+                <div class="row">
+                    <div class="dropdown docs-version-dropdown">
+                        <button class="btn btn-secondary btn-lg dropdown-toggle docs-version-dropdown-button" type="button" data-toggle="dropdown">
+                            Select Docs
+                        </button>
+                        <div class="dropdown-menu version-list">
+                            {% for version in docs-versions %}
+                                {% if version == docs-versions[0] %}    
```

Comment:
should use [foorloop-first](https://help.shopify.com/en/themes/liquid/objects/for-loops#forloop-first) here instead. I'm assuming docs-versions is already sorted with latest as first, right?